webapp/src/components/tutorial/TutorialStepCounter.tsx

```diff
@@ -87,7 +68,13 @@ export function TutorialStepCounter(props: TutorialStepCounterProps) {
                     label={stepNum === currentStep ? `${stepNum + 1}` : undefined}
                 />
             })}
-            {nextButton}
+            <Button
+                className={props.isHorizontal ? "ui button counter-next-button" : "square-button"}
+                leftIcon={`icon ${lastStep ? "check" : props.isHorizontal ? "arrow circle right" : "right chevron"}`}
+                onClick={lastStep ? props.onDone : handleNextStep}
+                aria-label={nextButtonTitle}
+                title={nextButtonTitle}
+                label={props.isHorizontal ? lastStep ? lf("Done") : lf("Next") : ""} />
```

Comment:
Is it possible to simplify this? Since we don't want to give a label if it's not horizontal, I'm wondering if we can just do something like 
`props.isHorizontal && lastStep ? lf("Done") : lf("Next")`

Typescript might get angry or it might cause bugs if `label` equals `null`. I'm not sure.

Reply:
I don't think there's much more simplification we can do here, unfortunately. In the sample you provided, we would end up with a "Next" label on the button when it's not-horizontal mode (i.e. in "Chevron" mode), but we actually want it to be no label in that case. We basically want no label in non-horizontal mode, then either "Done" or "Next" based in horizontal mode based on whether or not it's the last step.

I guess we could try setting the label as "Next" in chevron-mode but hide it through CSS instead...

Reply:
Oh, right. No, no. The way you have it is the most simple, you're right. I don't know why my brain defaulted to "if `isHorizontal` is false then nothing will happen". I guess I was trying to apply short circuiting here, but that's not a thing with ternaries.